client/client.go

```diff
@@ -22,57 +22,318 @@ limitations under the License.
+		return "", err
+	}
+	if isHost(id) {
+		if isRunningSystemd() {
+			subsystemPath = filepath.Join(groupPath, systemSlice)
+		} else {
+			subsystemPath = groupPath
+		}
+		return subsystemPath, nil
+	}
+
+	if isFsCgroupParent(groupPath) {
+		// default cgroupfs parent is used for container
+		subsystemPath = filepath.Join(groupPath, "docker", id)
+	} else {
+		// cgroup is created under systemd.slice
+		subsystemPath = filepath.Join(groupPath, systemSlice, "docker-"+id+".scope")
+	}
+
+	return subsystemPath, nil
+}
+
+// GetStatsFromContainer returns docker containers stats: cgroups stats (cpu usage, memory usage, etc.) and network stats (tx_bytes, rx_bytes etc.)
+// Notes: incoming container id has to be full-length to be able to inspect container
+func (dc *dockerClient) GetStatsFromContainer(id string, collectFs bool) (*wrapper.Statistics, error) {
+	var (
+		stats      = wrapper.NewStatistics()
+		groupWrap  = wrapper.Cgroups2Stats // wrapper for cgroup name and interface for stats extraction
+		err        error
+		workingSet uint64
+	)
+	container := &docker.Container{}
+
+	var pid int
+
+	if !isHost(id) {
+		if !isFullLengthID(id) {
+			return stats, fmt.Errorf("Container id %+v is not fully-length - cannot inspect container", id)
+		}
+		// inspect container based only on fully-length container id.
+		container, err = dc.InspectContainer(id)
+
+		if err != nil {
+			// only log error message and return stats (contains cgroups stats)
+			return stats, nil
+		}
+		pid = container.State.Pid
+	}
+
+	for cg, stat := range groupWrap {
+		var groupPath string
+		var err error
+
+		groupPath, err = GetSubsystemPath(cg, id)
+		if err != nil {
+			continue
+		}
+		// get cgroup stats for given docker
+		err = stat.GetStats(groupPath, stats.CgroupStats)
+		if err != nil {
+			// just log about it
+			if isHost(id) {
+				fmt.Fprintln(os.Stderr, "Cannot obtain cgroups statistics for host, err=", err)
+			} else {
+				fmt.Fprintln(os.Stderr, "Cannot obtain cgroups statistics for container: id=", id, ", image=", container.Image, ", name=", container.Name, ", err=", err)
+			}
+			continue
+		}
+	}
+
+	// calculate additional stats memory:working_set based on memory_stats
+	if totalInactiveAnon, ok := stats.CgroupStats.MemoryStats.Stats["total_inactive_anon"]; ok {
+		workingSet = stats.CgroupStats.MemoryStats.Usage.Usage
+		if workingSet < totalInactiveAnon {
+			workingSet = 0
+		} else {
+			workingSet -= totalInactiveAnon
+		}
+
+		if totalInactiveFile, ok := stats.CgroupStats.MemoryStats.Stats["total_inactive_file"]; ok {
+			if workingSet < totalInactiveFile {
+				workingSet = 0
+			} else {
+				workingSet -= totalInactiveFile
+			}
+		}
+	}
+	stats.CgroupStats.MemoryStats.Stats["working_set"] = workingSet
+
+	if !isHost(id) {
+		rootFs := "/"
+
+		stats.Network, err = network.NetworkStatsFromProc(rootFs, pid)
+
+		if err != nil {
+			// only log error message
```

Comment:
In this checking of `err` you put the comment in the next several checking of `err` you didn't put. This code looks similarly so I suggest either putting comments everywhere or removing of this one.